### ISSUE TITLE
Move implementation of BQL functions dealing with Beancount specific types to the `beancount` data source

### ISSUE THREAD

**Issue by dnicolodi** [beancount/beanquery#176]
The BQL core should not depend on Beancount.

**Comment by blais** [comment]
I agree. It would be nice if it optionally sourced custom aggregators and data types from the Beancount install when applied to Beancount, but could work independently otherwise.
